src/ejabberd_c2s.erl

```diff
@@ -2969,8 +2969,9 @@ flush_stream_mgmt_buffer(#state{stream_mgmt = StreamMgmt})
 flush_stream_mgmt_buffer(#state{stream_mgmt_buffer = Buffer} = State) ->
     re_route_packets(Buffer, State).
 
-re_route_packets(Buffer, StateData) ->
-    [reroute(Acc, StateData) || Acc <- lists:reverse(Buffer)],
+re_route_packets(Buffer, #state{jid = Jid, host_type = HostType} = StateData) ->
+    FilteredBuffer = mongoose_hooks:filter_unacknowledged_messages(HostType, Jid, Buffer),
```

Comment:
maybe a good idea to pass already reversed input into filter_unacknowledged_messages

Reply:
Good one